src/main/java/org/prebid/server/handler/NotificationEventHandler.java

```diff
@@ -0,0 +1,148 @@
+package org.prebid.server.handler;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.Json;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.analytics.AnalyticsReporter;
+import org.prebid.server.analytics.model.HttpContext;
+import org.prebid.server.analytics.model.NotificationEvent;
+import org.prebid.server.exception.InvalidRequestException;
+import org.prebid.server.proto.request.EventNotificationRequest;
+import org.prebid.server.util.ResourceUtil;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Accepts notifications from browsers and mobile application for further processing by {@link AnalyticsReporter}
+ * and responding with tracking pixel when requested.
+ */
+public class NotificationEventHandler implements Handler<RoutingContext> {
+
+    private static final String TRACKING_PIXEL_PNG = "static/tracking-pixel.png";
+    private static final String TRACKING_PIXEL_JPG = "static/tracking-pixel.jpg";
+    private static final String VIEW_TYPE = "view";
+    private static final String WIN_TYPE = "win";
+    private static final String FORMAT_PARAMETER = "format";
+    private static final String JPG = "jpg";
+    private static final String PNG = "png";
+    private static final String JPG_HEADER = "image/jpeg";
+    private static final String PNG_HEADER = "image/png";
+
+    private final AnalyticsReporter analyticsReporter;
+    private final byte[] trackingPixelPng;
+    private final byte[] trackingPixelJpg;
+
+    private NotificationEventHandler(AnalyticsReporter analyticsReporter, byte[] trackingPixelPng,
+                                     byte[] trackingPixelJpg) {
+        this.analyticsReporter = analyticsReporter;
+        this.trackingPixelJpg = trackingPixelJpg;
+        this.trackingPixelPng = trackingPixelPng;
+    }
+
+    public static NotificationEventHandler create(AnalyticsReporter analyticsReporter) {
+        return new NotificationEventHandler(Objects.requireNonNull(analyticsReporter),
+                readTrackingPixel(TRACKING_PIXEL_PNG), readTrackingPixel(TRACKING_PIXEL_JPG));
+    }
+
+    private static byte[] readTrackingPixel(String path) {
+        try {
+            return ResourceUtil.readByteArrayFromClassPath(path);
+        } catch (IOException e) {
+            throw new IllegalArgumentException(String.format("Failed to load pixel image at %s", path), e);
+        }
+    }
+
+    @Override
+    public void handle(RoutingContext context) {
+        final EventNotificationRequest eventNotificationRequest;
+        try {
+            eventNotificationRequest = makeEventRequest(context.getBody());
```

Comment:
I don't think that's correct - it is a `GET` request and all parameters should be read from query params. 

Fixed tech spec to explicitly say it is a `GET` request